app/utils/geometry.py

```
@@ -0,0 +1,39 @@
+            )
+        )
+        if intersects:
+            inside = not inside
+        j = i
+    return inside
+
+
+def centroid(polygon: Iterable[GeoPoint]) -> GeoPoint:
+    poly = list(polygon)
+    if not poly:
+        raise ValueError("Polygon must not be empty")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)